src/pages/teachers/specialist/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug]/downloads.tsx

```diff
@@ -0,0 +1,99 @@
+import {
+  NextPage,
+  GetStaticProps,
+  GetStaticPathsResult,
+  GetStaticPropsResult,
+} from "next";
+
+import {
+  getFallbackBlockingConfig,
+  shouldSkipInitialBuild,
+} from "@/node-lib/isr";
+import getPageProps from "@/node-lib/getPageProps";
+import curriculumApi2023 from "@/node-lib/curriculum-api-2023";
+import { SpecialistLessonDownloadsPageData } from "@/node-lib/curriculum-api-2023/queries/specialistLessonDownload/specialistLessonDownload.schema";
+import SpecialistLessonDownloads from "@/components/TeacherViews/SpecialistLessonDownloads/SpecialistLessonDownloads.view";
+import AppLayout from "@/components/SharedComponents/AppLayout";
+import { getSeoProps } from "@/browser-lib/seo/getSeoProps";
+
+const SpecialistLessonDownloadsPage: NextPage<
+  SpecialistLessonDownloadsPageData
+> = ({ curriculumData }) => {
+  const { lesson } = curriculumData;
+  return (
+    <AppLayout
+      seoProps={{
+        ...getSeoProps({
+          title: `Lesson Download: ${lesson.lessonTitle} | ${lesson.subjectTitle}`,
+          description: "Lesson downloads",
+        }),
+        ...{ noFollow: true, noIndex: true },
+      }}
+    >
+      <SpecialistLessonDownloads curriculumData={curriculumData} />
+    </AppLayout>
+  );
+};
+
+export type URLParams = {
+  lessonSlug: string;
+  programmeSlug: string;
+  unitSlug: string;
+};
+
+export const getStaticPaths = async () => {
+  if (shouldSkipInitialBuild) {
+    return getFallbackBlockingConfig();
+  }
+
+  const config: GetStaticPathsResult<URLParams> = {
+    fallback: "blocking",
+    paths: [],
+  };
+  return config;
+};
+
+export const getStaticProps: GetStaticProps<
+  SpecialistLessonDownloadsPageData,
+  URLParams
+> = async (context) => {
+  return getPageProps({
+    page: "specialist-downloads::getStaticProps",
+    context,
+    getProps: async () => {
+      if (!context.params) {
+        throw new Error("No context.params");
+      }
+      const { lessonSlug, programmeSlug, unitSlug } = context.params;
+
+      try {
+        const curriculumData =
+          await curriculumApi2023.specialistLessonDownloads({
+            programmeSlug,
+            unitSlug,
+            lessonSlug,
+          });
+
+        if (!curriculumData) {
```

Comment:
Does the try / catch not do the same thing twice, throwing an error if there's no `curriculumData`?

Reply:
the query itself can throw an error so we need to catch that and show the 404 page, whereas this just shows the 404 if the query succeeds but returns undefined/null

Reply:
ahh ok ok, I think I presumed that the query file would catch the error and throw it at that point